app/annotationsList/annotationList.js

```diff
@@ -142,22 +142,15 @@ app.controller('AnnotationListCtrl', function($rootScope, $scope, $http, $uibMod
 
   };
 
-  $scope.showAnnotationExtension = function(extension) {
-    $scope.extensions = [];
-    _.each(extension.split('|'), function(ext){
-      var extensionsLevel2 = [];
-      var extensionSplit = ext.split(',');
-      _.each(extensionSplit, function(extensionSplit){
-        var regExp = /([^(]+)\(([^)]+):([^)]+)\)/;
-        var match = regExp.exec(extensionSplit);
-        extensionsLevel2.push({
-          'relationship':match[1],
-          'database':match[2],
-          'id':match[3]
-        });
+  $scope.showAnnotationExtension = function(extensions) {
+    angular.forEach(extensions, function(extension){
+      angular.forEach(extension.connectedXrefs, function(xref){
+        olsService.getTermName(xref).then(function(name){
+          xref.label = name.data.label;
+        })
       });
-      $scope.extensions.push(extensionsLevel2);
     });
+    $scope.extensions = extensions;
```

Comment:
There are some unused variables and un-terminated statements that could be improved in this file.